File: Java/PrimeNumberChecker.java

import java.util.Scanner;

/**
 * @author dudmacedo
 */
public class PrimeNumberChecker {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            System.out.print("Enter a integer number to test if it is Prime (<= 1 to exit): ");

            Integer n = sc.nextInt();

            if (n <= 1) {
                System.out.println("Exiting...");
                return;
            }

            System.out.println(
                    (isPrime(n))
                    ? n + " is Prime!"
                    : n + " is Not Prime!");
        }

    }

    private static boolean isPrime(Integer n) {
        if (n <= 1) {
            return false;
        } else if (n == 2) {
            return true;
        } else if (n % 2 == 0) {
            return false;
        }

        int limit = (int) Math.sqrt(n) + 1;
        int factor = 3;

        while (factor < limit) {
            if (n % factor == 0) {
                return false;
            }
            factor += 2;
        }

        return true;
    }

}

File: Python/prime_number_checker.py
import re
import sys

from math import ceil, sqrt


def is_prime(n: int) -> bool:
    """Returns True iff n is prime."""
    if n == 2:
        return True
    elif n < 2 or n % 2 == 0:
        return False
    
    limit = ceil(sqrt(n))
    
    for factor in range(3, limit + 1, 2):
        if n % factor == 0:
            return False
    
    return True


def main():
    while True:
        num = input('Enter a number (or q to quit): ')
        if num.lower() in ['q', 'quit']:
            print('Bye!')
            sys.exit(0)
        if not re.match(r'^[-]?\d+$', num):
            print(f"'{num}' is not a valid number")
            sys.exit(1)
        
        primeness = is_prime(int(num))
        print(f"{num} is {'' if primeness else 'NOT '} prime")


if __name__ == '__main__':
    main()
